File: src/js/src/satisfiers/OpenNonNestableInlineScopeSatisfier.js
function OpenNonNestableInlineScopeSatisfier(scopeType) {
	var innerSatisfier = new NegatingSatisfier(new InScopeStackSatisfier(scopeType));
	
	this.isSatisfiedBy = function(context) {
		return innerSatisfier.isSatisfiedBy(context);
	};
}

File: src/dotnet/Src/ButterflyNet.Parser/Satisfiers/OpenNonNestableInlineScopeSatisfier.cs
using System;

namespace ButterflyNet.Parser.Satisfiers {
	public class OpenNonNestableInlineScopeSatisfier : ISatisfier {
		private readonly ISatisfier innerSatisfier;

		public OpenNonNestableInlineScopeSatisfier(Type type) {
			innerSatisfier = new NegatingSatisfier(new InScopeStackSatisfier(type));
		}

		public bool IsSatisfiedBy(ParseContext context) {
			return innerSatisfier.IsSatisfiedBy(context);
		}
	}
}